app/src/main/java/com/example/plantsapp/presentation/ui/authentication/AuthViewModel.kt

```diff
@@ -23,10 +23,14 @@ class AuthViewModel @Inject constructor(
         data class AuthError(@StringRes val errorId: Int) : AuthResult()
     }
 
+    private val _isLoading: MutableLiveData<Boolean> = MutableLiveData(false)
+    val isLoading: LiveData<Boolean> get() = _isLoading
+
     private val _authResult: MutableLiveData<Event<AuthResult>> = MutableLiveData()
     val authResult: LiveData<Event<AuthResult>> get() = _authResult
 
     fun onSignInResult(token: String?) {
+        _isLoading.value = true
```

Comment:
Here is one problem: 
What will happen if token == null? 

How about make loading more correct - 
start right before async work, 
and stop it after 


When you work with async work, 
it is better to use try-catch-finally
so in try you show loading
in finally you hide 

So it will guaranty us that there will never be situation when something happens and you will be with infinite loading

Reply:
fix this in other classes too